relayer/forwarder.go

```diff
@@ -0,0 +1,123 @@
+package relayer
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+
+	rcommon "github.com/athanorlabs/go-relayer/common"
+	"github.com/athanorlabs/go-relayer/impls/gsnforwarder"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+
+	contracts "github.com/athanorlabs/atomic-swap/ethereum"
+)
+
+func createForwarderSignature(
+	ctx context.Context,
+	claimerEthKey *ecdsa.PrivateKey,
+	ec *ethclient.Client,
+	relayerFeeWei *big.Int,
+	swapFactoryAddress ethcommon.Address,
+	forwarderAddress ethcommon.Address,
+	swap *contracts.SwapFactorySwap,
+	secret *[32]byte,
+) ([]byte, error) {
+
+	if swap.Claimer != ethcrypto.PubkeyToAddress(claimerEthKey.PublicKey) {
+		return nil, fmt.Errorf("signing key does not match claimer %s", swap.Claimer)
+	}
```

Comment:
this is interesting; I think there's an attack vector if someone else signs the `claimRelayer` function for someone else;
ie.
1. I want to steal someone's swap funds so I become an (evil) relayer and wait for someone to send me a tx to relay
2. The tx they send me has their swap secret.  I sign the `claimRelayer` transaction with `fee == swap.value` and the swap secret
3. I submit this to the contract via the forwarder contract
4. I claim all the swap funds

This can easily be mitigated by adding a check `_msgSender() == swap.claimer` inside `claimRelayer()`. There's no check for who actually signed the message inside `claimRelayer()` currently, which opens up this attack vector.

Normal non-relayer `claim()` doesn't have this issue unless someone gives out their swap secret, which they obviously shouldn't.

I will add this in :)

Reply:
Fix merged into master by https://github.com/AthanorLabs/atomic-swap/pull/348.